scipy/spatial/src/distance_impl.h

```diff
@@ -365,8 +365,14 @@ jensenshannon_distance_double(const double *p, const double *q, const npy_intp n
         const double m_i = (p_i + q_i) / 2.0;
         if (p_i > 0.0)
             s += p_i * log(p_i / m_i);
-        if (q_i > 0.0)
-            s += q_i * log(q_i / m_i);
+        if (q_i > 0.0) {
+            if (fpclassify(q_i) == FP_SUBNORMAL) {
+                s += q_i;
+            }
```

Comment:
This effectively amounts to "do nothing for subnormals" in this control flow I think (add in something that is basically zero). Per my original comment, should probably try moving the subnormal to the last element in the array as well, to see if i.e., `q_i` would also benefit from a similar shim (assuming this is what we actually want to do).